### PR TITLE
QCElemental 0.23.0 update

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR should make QCFractal compatible with qcelemental=0.23.0, we now use the json encoding on dicts when adding or updating results which should remove all non-json serializable objects. 

## Changelog description
Update QCFractal to work with QCElemental=0.23.0

## Status

- [x] Code base linted
- [ ] Ready to go

### CODE REVIEW

File: devtools/conda-envs/openff.yaml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 name: qcarchive
 channels:
   - defaults
-  - omnia
-  - psi4/label/dev
```
Comment (bennybp) [inline devtools/conda-envs/openff.yaml:5]: I think we should leave in psi4/label/dev and test against the current master version. Right @loriab ?
I suppose we can leave the pin below

**Comment by bennybp** [comment]
I have zero idea why this is failing now, and cannot reproduce locally

**Comment by bennybp** [comment]
Ok seems to be something with alembic on the conda defaults channel not pulling in importlib_resources. The reason for that is a conda-related mystery which I can't be bothered to figure out (since it does seem to be specified as a dependency).

If you install alembic from the default channel, and then do `pip install -e .` for qcfractal, it then gets installed because pip is smart enough to actually handle dependencies I guess. (but we do `--no-deps` in the CI)

**Comment by bennybp** [comment]
Ok ready to merge if you are ok with the psi4 version change

**Comment by jthorton** [comment]
@bennybp thanks for fixing this, LGTM!

**Review by bennybp — APPROVED** [review]

**Pull request merged**